zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -3256,7 +3256,7 @@ pub fn pczt_single_step<P0: ShieldedPoolTester, P1: ShieldedPoolTester, DSF>(
 pub fn wallet_recovery_computes_fees<T: ShieldedPoolTester, DsF: DataStoreFactory>(
     ds_factory: DsF,
     cache: impl TestCache,
-    intervene: impl FnOnce(&mut DsF::DataStore, TxId) -> Result<(), DsF::DsError>,
+    intervene: &mut impl FnMut(&mut DsF::DataStore, TxId) -> Result<(), DsF::DsError>,
```

Comment:
An `&mut` should be unnecessary here. If Rust is requiring `intervene` to be mutable, `mut impl FnMut(_)` is the way to do it.